### PR TITLE
Fix code style issues in MacOS embedder

### PR DESCRIPTION
**Pull request by knopp**
## Description

PR to fix code style issues introduced in https://github.com/flutter/engine/pull/21525.

## Related Issues

https://github.com/flutter/flutter/issues/44136

## Tests

I added the following tests:

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [X] I read the [contributor guide] and followed the process outlined there for submitting PRs.
- [X] I signed the [CLA].
- [X] I read and followed the [C++, Objective-C, Java style guides] for the engine.
- [X] I read the [tree hygiene] wiki page, which explains my responsibilities.
- [X] I updated/added relevant documentation.
- [X] All existing and new tests are passing.
- [X] I am willing to follow-up on review comments in a timely manner.

## Reviewer Checklist

- [ ] I have submitted any presubmit flakes in this PR using the [engine presubmit flakes form] before re-triggering the failure.

## Breaking Change

Did any tests fail when you ran them? Please read [handling breaking changes].

- [X] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*

[issue database]: https://github.com/flutter/flutter/issues
[contributor guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[tree hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[engine presubmit flakes form]: https://forms.gle/Wc1VyFRYJjQTH6w5A

### CODE REVIEW

File: shell/platform/darwin/macos/BUILD.gn [hunk 1]
```diff
@@ -66,14 +66,13 @@ source_set("flutter_framework_source") {
...
-    "framework/Source/MacOSSwitchableGLContext.mm",
+    "framework/Source/MacOSGLContextSwitch.h",
+    "framework/Source/MacOSGLContextSwitch.mm",
   ]
 
   sources += _flutter_framework_headers
 
   deps = [
-    "//flutter/flow:flow",
     "//flutter/fml:fml",
```
Comment (iskakaushik) [inline shell/platform/darwin/macos/BUILD.gn:76]: I think we don't want to depend on `fml` as well.

Comment (knopp) [reply shell/platform/darwin/macos/BUILD.gn]: Right. Removed.

Comment (stuartmorgan) [reply shell/platform/darwin/macos/BUILD.gn]: FML is actually okay; we're going to need it for a11y support, so I've looked into it already, and it's generally very base-level, with minimal dependencies.

Comment (knopp) [reply shell/platform/darwin/macos/BUILD.gn]: In this instance it was only used for single FML_DCHECK call, which I've replaced with NSAssert. I can put it back if FML_DCHECK is better.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Some small nits, but overall this looks great. Thanks so much for the quick follow up!

File: shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
 
 #include <OpenGL/gl.h>
+#import "flutter/shell/platform/darwin/macos/framework/Source/MacOSGLContextSwitch.h"
 
 enum {
-  kFront = 0,
-  kBack = 1,
+  kFrontBuffer = 0,
+  kBackBuffer = 1,
   kBufferCount,
```
[the remark this answers is not in the document]
Comment (knopp) [reply shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm:13]: Sure. I looked at the constants elsewhere (i.e. FlutterTextInputPlugin) which don't seem to be prefixed, but I don't really mind.

**Comment by knopp** [comment]
@stuartmorgan, these should be fixed now.

File: shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm (hunk 2 above)
[the remark this answers is not in the document]
Comment (stuartmorgan) [reply shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm]: Well, the style guide doesn't require prefixing for non-public constants. But my concern here is that it's much more likely that someone would accidentally not prefix a public constant (as happened with some very unfortunately generic constants in embedder.h, for instance) called `kBufferCount` than, say `kClearClientMethod`.

[the remark this answers is not in the document]
Comment (knopp) [reply shell/platform/darwin/macos/framework/Source/FlutterSurfaceManager.mm]: No argument here really, fixed in [cfbe14bc6](https://github.com/flutter/engine/pull/22270/commits/cfbe14bc67039537f18f2c14152729437f006d29#diff-3d541252e6b0c40586936e8d15b3e6114053541631c9f9d721e7353fdf17cecaR12).

**Review by stuartmorgan — APPROVED** [review]

**Comment by stuartmorgan** [comment]
Thanks!

**Pull request merged**